arrow-fx-stm/src/main/kotlin/arrow/fx/stm/TMVar.kt

```diff
@@ -19,6 +19,184 @@ fun <A> STM.newEmptyTMVar(): TMVar<A> = TMVar<A>(newTVar(Option.None))
  * - [TMVar.new] and [STM.newTMVar] create a new filled [TMVar]
  * - [TMVar.empty] and [STM.newEmptyTMVar] create an empty [TMVar]
  *
+ * ## Reading the content of a [TMVar]
+ *
+ * Taking the value out of a [TMVar]:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.stm.TMVar
+ * import arrow.fx.stm.atomically
+ * import arrow.fx.coroutines.Environment
+ *
+ * suspend fun main() {
+ *   Environment().unsafeRunSync {
+ *     //sampleStart
+ *     val tmvar = TMVar.new(10)
+ *     val result = atomically {
+ *       tmvar.take()
+ *     }
+ *     //sampleEnd
+ *     println("Result $result")
+ *     println("New value ${atomically { tmvar.tryTake() } }")
+ *   }
+ * }
+ * ```
+ *
+ * Should the [TMVar] be empty at the time of calling [STM.take], it will call [STM.retry], suspend and wait for another transaction to
```

Comment:
Should be added to the definition of `take` as well.

Reply:
It already is, though with different wording.